bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-list-item/user-dropdown/component.jsx

```diff
@@ -0,0 +1,519 @@
+
+const messages = defineMessages({
+  presenter: {
+    id: 'app.userList.presenter',
+    description: 'Text for identifying presenter user',
+  },
+  you: {
+    id: 'app.userList.you',
+    description: 'Text for identifying your user',
+  },
+  locked: {
+    id: 'app.userList.locked',
+    description: 'Text for identifying locked user',
+  },
+  guest: {
+    id: 'app.userList.guest',
+    description: 'Text for identifying guest user',
+  },
+  menuTitleContext: {
+    id: 'app.userList.menuTitleContext',
+    description: 'adds context to userListItem menu title',
+  },
+  userAriaLabel: {
+    id: 'app.userList.userAriaLabel',
+    description: 'aria label for each user in the userlist',
+  },
+  statusTriggerLabel: {
+    id: 'app.actionsBar.emojiMenu.statusTriggerLabel',
+    description: 'label for option to show emoji menu',
+  },
+  backTriggerLabel: {
+    id: 'app.audio.backLabel',
+    description: 'label for option to hide emoji menu',
+  },
+  ChatLabel: {
+    id: 'app.userList.menu.chat.label',
+    description: 'Save the changes and close the settings menu',
+  },
+  ClearStatusLabel: {
+    id: 'app.userList.menu.clearStatus.label',
+    description: 'Clear the emoji status of this user',
+  },
+  MakePresenterLabel: {
+    id: 'app.userList.menu.makePresenter.label',
+    description: 'Set this user to be the presenter in this meeting',
+  },
+  RemoveUserLabel: {
+    id: 'app.userList.menu.removeUser.label',
+    description: 'Forcefully remove this user from the meeting',
+  },
+  MuteUserAudioLabel: {
+    id: 'app.userList.menu.muteUserAudio.label',
+    description: 'Forcefully mute this user',
+  },
+  UnmuteUserAudioLabel: {
+    id: 'app.userList.menu.unmuteUserAudio.label',
+    description: 'Forcefully unmute this user',
+  },
+  PromoteUserLabel: {
+    id: 'app.userList.menu.promoteUser.label',
+    description: 'Forcefully promote this viewer to a moderator',
+  },
+  DemoteUserLabel: {
+    id: 'app.userList.menu.demoteUser.label',
+    description: 'Forcefully demote this moderator to a viewer',
+  },
+});
+
+const propTypes = {
+  compact: PropTypes.bool.isRequired,
+  user: PropTypes.shape({}).isRequired,
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }).isRequired,
+  normalizeEmojiName: PropTypes.func.isRequired,
+  meeting: PropTypes.shape({}).isRequired,
+  isMeetingLocked: PropTypes.func.isRequired,
+  getScrollContainerRef: PropTypes.func.isRequired,
+};
+
+class UserDropdown extends Component {
+  /**
+   * Return true if the content fit on the screen, false otherwise.
+   *
+   * @param {number} contentOffSetTop
+   * @param {number} contentOffsetHeight
+   * @return True if the content fit on the screen, false otherwise.
+   */
+  static checkIfDropdownIsVisible(contentOffSetTop, contentOffsetHeight) {
+    return (contentOffSetTop + contentOffsetHeight) < window.innerHeight;
+  }
+
+  static createAction(action, ...options) {
+    return (
+      <UserAction
```

Comment:
Does this `UserAction` is actually needed? Some places you are using `DropdownListItem` and others the wrapper (`UserAction`). Please choose one and stick with it.

My suggestion would be to stick with `DropdownListItem`.